File: initializers/src/main/java/org/springframework/boot/autoconfigure/mustache/MustacheInitializer.java
package org.springframework.boot.autoconfigure.mustache;

import com.samskivert.mustache.Mustache;

import org.springframework.context.ApplicationContextInitializer;
import org.springframework.context.support.GenericApplicationContext;

/**
 * {@link ApplicationContextInitializer} adapter for {@link MustacheAutoConfiguration}.
 */
public class MustacheInitializer implements ApplicationContextInitializer<GenericApplicationContext> {

	private MustacheProperties properties;

	public MustacheInitializer(MustacheProperties properties) {
		this.properties = properties;
	}

	@Override
	public void initialize(GenericApplicationContext context) {
		MustacheAutoConfiguration configuration = new MustacheAutoConfiguration(this.properties, context.getEnvironment(), context);
		context.registerBean(MustacheResourceTemplateLoader.class, () -> configuration.mustacheTemplateLoader());
		context.registerBean(Mustache.Compiler.class, () -> configuration.mustacheCompiler(context.getBean(Mustache.TemplateLoader.class)));
	}
}

File: kofu/src/main/kotlin/org/springframework/boot/autoconfigure/mustache/MustacheInitializer.kt
package org.springframework.boot.autoconfigure.mustache

import org.springframework.beans.factory.getBean
import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.support.GenericApplicationContext
import org.springframework.context.support.registerBean

internal class MustacheInitializer(private val properties: MustacheProperties) : ApplicationContextInitializer<GenericApplicationContext> {

	override fun initialize(context: GenericApplicationContext) {
		val mustacheConfiguration = MustacheAutoConfiguration(properties, context.environment, context)
		context.registerBean {
			mustacheConfiguration.mustacheTemplateLoader()
		}
		context.registerBean {
			mustacheConfiguration.mustacheCompiler(context.getBean())
		}
		val mustacheReactiveConfiguration = MustacheReactiveWebConfiguration(properties)
		context.registerBean {
			mustacheReactiveConfiguration.mustacheViewResolver(context.getBean())
		}
	}
}